File: priv/javascript/lib/integer.js
import Kernel from './kernel';

let Integer = {

  is_even: function(n){
    return n % 2 === 0;
  },

  is_odd: function(n){
    return n % 2 !== 0;
  },

  parse: function(bin){
    let result = parseInt(bin);

    if(isNaN(result)){
      return Kernel.SpecialForms.atom("error");
    }

    let indexOfDot = bin.indexOf(".");

    if(indexOfDot >= 0){
      return Kernel.SpecialForms.tuple(result, bin.substring(indexOfDot));
    }

    return Kernel.SpecialForms.tuple(result, "");
  },

  to_char_list: function(number, base = 10){
    return number.toString(base).split('');
  },

  to_string: function(number, base = 10){
    return number.toString(base);
  }
};

export default Integer;

File: priv/std_lib/integer.ex
defmodule ElixirScript.Integer do
  @moduledoc false
  def is_even(number) do
    rem(number, 2) == 0
  end

  def is_odd(number) do
    rem(number, 2) != 0
  end

  def to_char_list(number, base \\ 10) do
    number.toString(base).split('')
  end

  def parse(bin, base \\ 10) do
    result = JS.parseInt(bin, base)

    if JS.isNaN(result) do
      :error
    else
      case bin.indexOf(".") do
        index_of_dot when index_of_dot < 0 ->
          {result, ""}
        index_of_dot ->
          {result, bin.substring(index_of_dot)}
      end
    end
  end
end
